beacon-chain/rpc/eth/node/handlers.go

```diff
@@ -37,14 +38,29 @@ func (s *Server) GetSyncStatus(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	isElOnline := s.ExecutionChainInfoFetcher.ExecutionClientConnected()
+	if !isElOnline {
+		isSyncing, err := s.ExecutionChainInfoFetcher.IsExecutionClientSyncing(ctx)
+		if !isSyncing {
+			if err != nil {
+				log.WithField("error", err.Error()).Debug("failed to get execution client sync status")
+			} else {
+				log.Debug("execution client is not syncing and not connected")
+			}
+		}
+		if isSyncing && err == nil {
+			isElOnline = true
+		}
```

Comment:
Why are we logging instead of returning this info to the client via `httputil.HandleError`?

Reply:
Also the error check can be the first thing you check, so that you don't have nesting

Reply:
fixed with sammy's suggestion